apps/transport/lib/jobs/consolidate_lez_job.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Transport.Jobs.ConsolidateLEZsJob do
+  @moduledoc """
+  Consolidates a low emission zones' national database
+  using valid `etalab/schema-zfe` resources published
+  on our platform.
+
+  Preferred way to run for now (until we have more safeguards!):
+  - in iex, run `consolidate()`, look at logs and inspect the files
+  - run the Oban job to consolidate and update files on data.gouv.fr
+  """
+  use Oban.Worker, max_attempts: 3
+  require Logger
+  import Ecto.Query
+  alias DB.{Dataset, Repo, Resource, ResourceHistory}
+
+  @schema_name "etalab/schema-zfe"
+  @lez_dataset_type "low-emission-zones"
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{}) do
+    consolidate() |> update_files()
+    :ok
+  end
+
+  def consolidate do
+    relevant_resources()
+    |> Enum.group_by(&type/1)
+    |> Enum.map(fn {type, resources} ->
+      details = resources |> Enum.map_join(", ", &"#{&1.dataset.custom_title} (#{&1.title})")
+      Logger.info("Found #{Enum.count(resources)} resources for #{type}: #{details}")
+      {type, consolidate_features(resources)}
+    end)
+  end
+
+  def update_files(consolidated_data) do
+    consolidated_data
+    |> Enum.each(fn {type, data} ->
+      filename = "#{type}.geojson"
+      path = write_file(filename, data |> Jason.encode!())
+
+      Datagouvfr.Client.Resources.update(%{
+        "dataset_id" => Map.fetch!(consolidation_configuration(), :dataset_id),
+        "resource_id" => resource_id_for_type(type),
+        "resource_file" => %{path: path, filename: filename}
+      })
+
+      Logger.info("Updated #{filename} on data.gouv.fr")
+
+      File.rm!(path)
```

Comment:
Suggestion: wrapper l'appel  à `update` dans un `try/after` (https://elixir-lang.org/getting-started/try-catch-and-rescue.html#after), pour être sûr d'effacer le fichier.

Ca peut éviter dans le futur des surprises (fichiers qui s'empilent).